Student-Unity/src/main/java/csed/swe/studentunity/DAO/UnverifiedUserRepo.java

```diff
@@ -0,0 +1,12 @@
+package csed.swe.studentunity.DAO;
+
+import csed.swe.studentunity.model.UnverifiedUser;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.Optional;
+
+
+public interface UnverifiedUserRepo extends JpaRepository<UnverifiedUser, Long> {
+    Optional<UnverifiedUser> findUnverifiedUserByEmail(String email);
```

Comment:
we made a primary key to the user instead of it's email, you should reference the user now by it's **id** not email for consistency.

Reply:
But unverified user table is for users who didn't verify submit a code sent to his email.
so when a user gets the verification code from his email and he submit this code the only thing that i have to identify this user in the table is using his email.
we use user id when a user has already logged in as i save his id in session id. 
but in this situation this is for unverified user.